generator/config/stage/merge.yaml

```diff
@@ -44,3 +45,136 @@ arguments:
         optional: true
         description: |
             The behavior of $merge if a result document does not match an existing document in the out collection.
+tests:
+    -
+        name: 'On-Demand Materialized View Initial Creation'
+        link: 'https://www.mongodb.com/docs/manual/reference/operator/aggregation/merge/#on-demand-materialized-view--initial-creation'
+        pipeline:
+            -
+                $group:
+                    _id:
+                        fiscal_year: '$fiscal_year'
+                        dept: '$dept'
+                    salaries:
+                        $sum: '$salary'
+            -
+                $merge:
+                    into:
+                        db: 'reporting'
+                        coll: 'budgets'
+                    on: '_id'
+                    whenMatched: 'replace'
+                    whenNotMatched: 'insert'
```

Comment:
2 enums should be created for this values.

Reply:
Should this be ticketed under [PHPLIB-1381](https://jira.mongodb.org/browse/PHPLIB-1381)?